source/core/slang-riff.h

```diff
@@ -17,6 +17,59 @@ struct RiffFourCC
      static const uint32_t kRiff = SLANG_FOUR_CC('R', 'I', 'F', 'F');
 };
 
+// Follows semantic version rules
+// https://semver.org/
+// 
+// major.minor.patch
+// Patch versions indicate a change. 
+// Minor means a change that is backwards compatible with previous minor versions. A step in minor and/or major zeros patch.
+// Major means a non compatible change. A step in major, zeros minor and patch. 
+struct RiffSemanticVersion
+{
+    typedef RiffSemanticVersion ThisType;
+    typedef uint32_t RawType;
+
+        /// ==
+    bool operator==(const ThisType& rhs) const { return m_raw == rhs.m_raw; }
+    bool operator!=(const ThisType& rhs) const { return !(*this == rhs); }
+    
+        /// A patch change indices a different version but does not change the compatibility of the format
+    int getPatch() const { return m_raw & 0xff; }
+        /// A minor change implies a format change that is backwards compatible  
+    int getMinor() const { return (m_raw >> 8) & 0xff; }
+        /// A major change is binary incompatible by default
+    int getMajor() const { return (m_raw >> 16); }
```

Comment:
What is the rationale for allocating the bits as 16 for major and then 8 each for minor and patch? This isn't a huge deal, but I could imagine somebody arguing that the patch field should have the most bits, or even that the memory savings here are so small that giving each field a full 16 or 32 bits isn't a real problem.

Reply:
I mean that could be argued. I think in practice it won't be a problem, because the patch part isn't really necessary for the serialization. If the patch came from some automated mechanism, then this would make more sense. 

A different way of dealing with this would be for the header to hold the git tag version for example.